extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php

```diff
@@ -1,14 +1,30 @@
 <?php
 
-class SiteWideMessagesController extends WikiaController  {
+class SiteWideMessagesController extends WikiaController {
 
 	const CACHE_VALIDITY_VARNISH = 10800; // 3 hours
 	const CACHE_VALIDITY_BROWSER = 3600; // 1 hour
 
+	private function shouldShowSiteWideMessage() {
+		global $wgEnableArticleFeaturedVideo;
+
+		$title = Title::newFromID( $this->getVal( 'articleId' ) );
+		if ( !is_null( $title ) ) {
+			$isFeaturedVideoEmbedded = !empty( $wgEnableArticleFeaturedVideo ) &&
+				ArticleVideoContext::isFeaturedVideoEmbedded( $title->getPrefixedDBkey() );
+		} else {
+			$isFeaturedVideoEmbedded = false;
+		}
+
+		return $this->wg->User->isLoggedIn() || $isFeaturedVideoEmbedded;
+	}
+
 	public function getAnonMessages() {
-		if ( $this->wg->User->isLoggedIn() ) {
+
+		if ( $this->shouldShowSiteWideMessage() ) {
```

Comment:
the method name says something opposite than it does ;) please change it to shouldNotShowSiteWideMessages()